src/shims/fs.rs

```diff
@@ -582,6 +597,256 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.set_last_error(ebadf)?;
         Ok((-1).into())
     }
+
+
+    fn mkdir(
+        &mut self,
+        path_op: OpTy<'tcx, Tag>,
+        mode_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.check_no_isolation("mkdir")?;
+
+        let mode = if this.tcx.sess.target.target.target_os.to_lowercase() == "macos" {
+            this.read_scalar(mode_op)?.not_undef()?.to_u16()? as u32
+        } else {
+            this.read_scalar(mode_op)?.to_u32()?
+        };
+
+        let path = this.read_os_str_from_c_str(this.read_scalar(path_op)?.not_undef()?)?;
+
+        let mut builder = DirBuilder::new();
+        #[cfg(target_family = "unix")]
```

Comment:
This looks suspicious, we are doing *something* based on the host platform... I suppose the fallback is somehow less good? Please add a comment to explain.

Also please make the scope of the `cfg`-then-else clearer, by adding a newline after `let _mode = mode;`.